docs/routing.md

```diff
@@ -262,3 +263,42 @@ app = Router(routes=[
     ])
 ])
 ```
+
+## WebSocket Routing
+
+When working with WebSocket endpoints, you should use `WebSocketRoute`
+instead of the usual `Route`.
+
+Path parameters, and reverse URL lookups for `WebSocketRoute` work the the same
+as HTTP `Route`, which can be found in the [Route](#path-parameters) documentation.
+
+```python
+from starlette.applications import Starlette
+from starlette.routing import WebSocketRoute
+
+
+async def websocket_index(websocket):
+    await websocket.accept()
+    await websocket.send_text("Hello, websocket!")
+    await websocket.close()
+
+
+async def websocket_user(websocket):
+    name = websocket.path_params["name"]
+    await websocket.accept()
+    await websocket.send_text(f"Hello, {name}")
+    await websocket.close()
+
+
+routes = [
+    WebSocketRoute("/", endpoint=websocket_index),
+    WebSocketRoute("/{name}", endpoint=websocket_user),
+]
+
+app = Starlette(routes=routes)
+```
+
+The `endpoint` argument can be one of:
+
+* An async function, which accepts a single `websocket` argument.
```

Comment:
We don't mention the `WebSocket` class on this page, should we?

Reply:
I think it would make sense. We don't do this for HTTP endpoints that accept a `Request` either. I think they can be helpful.

Reply:
After giving it more thought I think there are lots of other places we can specif technical details, probably it can be a better option to add docstrings or something.
I'll post this into gitter to get some feedback.

Reply:
So... You want to wait something before merging this?

Reply:
Already tried that, I don't think we can do that. IMO this should be good for now.